### DISCUSSION TITLE
importJWK does not work on unsecure domain

### DISCUSSION THREAD

**Discussion by krystianknowak** [panva/jose#547] [Q&A] [answered]
Hello,

I have a case when I need to use `importJWK` function outside secure domain, in the browser.
When I call `importJWK` in my app on localhost, or domain with `https` everything works as it should, but on unsecure domains which for me is just remote host without https `importJWK` throws an error
![image](https://github.com/panva/jose/assets/28805489/1038d482-f358-4432-b4bd-1ada5a4aee4b)
I found out that this is related to `Web Crypto Api` and that `subtle` property is not available on unsecure domains.
What I want to do is to importJWK on unsecure domain(or just read publicKey from it). Is there a way to omit using web crypto api in any way? 
Code:
![image](https://github.com/panva/jose/assets/28805489/239bb0b4-7ed9-4de2-8629-391f9b7ac937)

**Comment by panva** [comment, marked as answer]
> Is there a way to omit using web crypto api in any way?

There is not, you need secure context. https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

**Reply by krystianknowak** [reply]
So, it would be possible only with different library which is not using `WebCrypto api` for jwt veryfication, yes?
